m3u2xspf.py

```diff
@@ -29,16 +32,33 @@ def get_tags(path):
     else:
         return {'location': path}
 
+import re
 
 def iter_lines(path):
     root = os.path.dirname(path)
+    track= {}
+    target = 'M3U'
     with open(sys.argv[1]) as fh:
         for line in fh:
             line = line.rstrip()
-            if line.startswith('#'):
+            if line.startswith('#EXTM3U'):
+                 target = 'M3UEXT'   
+            elif line.startswith('#EXTINF'):
+                matches = re.compile('#EXTINF:(\d*),([^,]+),(?:logo=(.*))?', re.IGNORECASE).findall(line) 
+                duration, title, logo =zip(matches[0])
+                track= { 
+                        'title' : title[0], 
+                        'image' : logo[0],
+                        'duration' : str(int(duration[0])*1000)
+                        }
+            elif line.startswith('#'):
                 pass
             elif line.startswith('http'):
-                yield {'location': line}
+                if target == 'M3UEXT':
+                    track['location'] = line
+                else:
+                    track={'location': line}
+                yield track
```

Comment:
We need to reset `track` to an empty dict here. Consider the following (invalid) input:

```
#EXTM3u
#EXTINF:686,China Light Parade in Strasbourg during the 2022 Christmas Market
https://www.youmaker.com/v/2GJoyNg88o76
https://www.youmaker.com/v/c7aj3asdnc83
```

If we don't reset `track` here, the metadata will be reused for the second entry. Following Postel's Law we should parse this as good as we can.